action/metadata-support.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/sethvargo/go-githubactions"
+	"golang.org/x/oauth2"
+)
+
+// getArtifacts queries github for artifact list, returning a 2 dimensional slice of
+// artifact names paired with their variant type.
+// org, repo, and workflowrunID are the github vars passed in to the API for the
+// workflow run query, see: https://docs.github.com/en/rest/actions/artifacts#list-workflow-run-artifacts
+func getArtifacts(org string, repo string, workflowRunID int64) [][]string {
+
+	// Auth to github API.
+	token := os.Getenv("CRT_GITHUB_TOKEN")
+	if token == "" {
+		githubactions.Errorf("missing env CRT_GITHUB_TOKEN")
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	ghClient := github.NewClient(tc)
+
+	// Query API for list of artifacts associated with specified build workflow of product.
+	artifacts, _, err := ghClient.Actions.ListWorkflowRunArtifacts(ctx, org, repo, workflowRunID, &github.ListOptions{})
+	if err != nil {
+		githubactions.Errorf(err)
+	}
+
+	// Extract a list of artifact names from the artifact data structure returned from the
+	// github API.
+	var rawArtifacts []string
+	for i := range artifacts.Artifacts {
+		rawArtifacts = append(rawArtifacts, *artifacts.Artifacts[i].Name)
+	}
+
+	// Parse raw list of artifact names into struct of Artifacts and variant names.
+	// For Vault products, associate variant types with their artifacts. All other products
+	// do not require variant grouping and thus are associated with variant type "all".
+	var processedArtifacts [][]string
```

Comment:
If we change the return type this will also need to be changed

```suggestion
        processedArtifacts := make(map[string][]string)
```